v2/client.go

```diff
@@ -478,15 +478,18 @@ func (c *Client) callAPI(ctx context.Context, r *request, opts ...RequestOption)
 			err = cerr
 		}
 	}()
-	c.debug("response: %#v", res)
-	c.debug("response body: %s", string(data))
-	c.debug("response status code: %d", res.StatusCode)
+	c.debug("response: %#v\n", res)
+	c.debug("response body: %s\n", string(data))
+	c.debug("response status code: %d\n", res.StatusCode)
 
 	if res.StatusCode >= http.StatusBadRequest {
 		apiErr := new(common.APIError)
 		e := json.Unmarshal(data, apiErr)
 		if e != nil {
-			c.debug("failed to unmarshal json: %s", e)
+			c.debug("failed to unmarshal json: %s\n", e)
+		}
+		if !apiErr.IsValid() {
+			return nil, fmt.Errorf("status_code=%v body=%v", res.StatusCode, string(data))
```

Comment:
`c.debug` is mostly used for debugging at the beginning of development by users, including the package developers. It is not typically used in the production environment. Therefore, it looks good if it returns an error. Adding a Response field with a type of `streaming/IO` is not good because the data has already been read, and the initial `stream/IO` has been fully consumed. So, I think its type should be `string` or `[]byte`. What do you think about this?

Reply:
> `c.debug` is mostly used for debugging at the beginning of development by users, including the package developers. It is not typically used in the production environment. Therefore, it looks good if it returns an error. Adding a Response field with a type of `streaming/IO` is not good because the data has already been read, and the initial `stream/IO` has been fully consumed. So, I think its type should be `string` or `[]byte`. What do you think about this?

I agree, let’s use `[]byte`, please only set this field when there is an error found, otherwise it will impact the performance of successful request.

Reply:
okay, had committed, do a review for last version.